src/Parser.y

```diff
@@ -608,6 +610,22 @@ checkConstantReassignment e = case G.expAst e of
   G.IndexAccess gId _ -> checkConstantReassignment gId
   _ -> return ()
 
+checkIterVariable :: G.Id -> G.CodeBlock -> ST.ParserMonad ()
+checkIterVariable id@(G.Id tk) (G.CodeBlock insts) = do
+  let assignments = filter isAssignment insts
+  let errors = filter modifiesIterVariable assignments
+  if length errors == 0
+  then return ()
+  else do
+    let (G.InstAsig f _) = head errors
+    let faultyTk = G.expTok f
+    RWS.tell [ST.SemanticError ("Iteration variable " ++ show id ++ " must not be reassigned") faultyTk]
+    return ()
+  where isAssignment (G.InstAsig _ _) = True
+        isAssignment _ = False
+        modifiesIterVariable (G.InstAsig l _) = (T.cleanedString (G.expTok l)) == (T.cleanedString tk)
+        modifiesIterVariable _ = False
+
```

Comment:
This works but it's not the best solution. The main problem is that iterator variables are not being inserted to symtable. We need to do so and mark the variable as a constant or something, so when we parse the CODEBLOCK of the for structure it checks that the variable reassigned is indeed a iterator variable and raise an error if so.

Reply:
with this program:
```
hello ashen one

traveling somewhere

   upgrading i with 1 soul until level 20
      traveling somewhere
         with orange saponite say i \
         with orange saponite say |\n|
      you died
   max level reached

you died

farewell ashen one
```

the program fails on the 7th line, pointing to the `i` variable usage though its a valid program

Reply:
with this program:
```
hello ashen one

traveling somewhere

   upgrading i with 1 soul until level 20
      traveling somewhere
         with orange saponite say i \
         with orange saponite say |\n|
      you died
   max level reached

you died

farewell ashen one
```

the program fails on the 7th line, pointing to the `i` variable usage though its a valid program

Reply:
![image](https://user-images.githubusercontent.com/29154504/72231657-16f76c80-3593-11ea-87ac-867bbadbbb3e.png)